src/lib/date.ts

```diff
@@ -0,0 +1,29 @@
+import { isDate, isValid, isLastDayOfMonth, differenceInCalendarDays } from "date-fns";
+
+/**
+ * Check whether the date is valid
+ * @param date The date
+ * @returns A boolean indicating whether the date is valid
+ */
+export function dateIsValid(date: Date): boolean {
+  return isDate(date) && isValid(date);
+}
+
+/**
+ * Check whether the date is the end of the month
+ * @param date The date
+ * @returns A boolean indicating whether the date is the end of the month
+ */
+export function dateIsEndOfMonth(date: Date): boolean {

```

Comment:
end of month is a bit vague, is last day of month would be more precise.

Reply:
Renamed to match the name of the date-fns function to be more precise 👍🏼